drivers/tee/optee/optee.c

```diff
@@ -874,15 +967,93 @@ static int optee_shm_unregister(const struct device *dev, struct tee_shm *shm)
 	return rc;
 }
 
-static int optee_suppl_recv(const struct device *dev, uint32_t func, unsigned int num_params,
+static int optee_suppl_recv(const struct device *dev, uint32_t *func, unsigned int *num_params,
 			    struct tee_param *param)
 {
+	struct optee_driver_data *data = (struct optee_driver_data *)dev->data;
+	struct optee_supp *supp = &data->supp;
+	struct optee_supp_req *req = NULL;
+
+	while (true) {
+		k_mutex_lock(&supp->mutex, K_FOREVER);
+		req = (struct optee_supp_req *)sys_dlist_peek_head(&supp->reqs);
+
+		if (req && supp->current == NULL) {
```

Comment:
so, can be a situation when you have `req != NULL` and `supp->current != NULL`
than the below check:
```
if (req)
```
finishes the loop 

is it what we want ?